samples/net/openthread/coap_client/src/coap_client.c

```diff
@@ -0,0 +1,174 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(coap_client, LOG_LEVEL_DBG);
+
+#include <logging/log_ctrl.h>
+#include <errno.h>
+#include <zephyr.h>
+#include <net/socket.h>
+
+#include "board_utils.h"
+#include "coap_utils.h"
+
+#define SLEEP_TIME 500
+#define IPV6_ADDR_LENGHT 16
+
+/**@brief Enumeration describing light commands. */
+typedef enum {
+	THREAD_COAP_UTILS_LIGHT_CMD_OFF = '0',
+	THREAD_COAP_UTILS_LIGHT_CMD_ON,
+	THREAD_COAP_UTILS_LIGHT_CMD_TOGGLE
+} thread_coap_utils_light_command_t;
+
+static struct k_work localLightWork;
+static struct k_work meshLightWork;
+static struct k_work meshProvisioningWork;
+char peerAddr[IPV6_ADDR_LENGHT + 1] = {
+	0,
+};
+
+static void wait_for_log_processing(void)
+{
+	while (log_buffered_cnt() > 0) {
```

Comment:
this seems like some kind of a hack? Is this really needed? Is there no other way?

Reply:
Removed. It is not needed any more.